File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/IAgentToolPermissionRepository.java
package com.enterpriseagentos.backendjava.domain.features.agents;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.YearMonth;
import java.util.*;
import java.util.concurrent.CompletableFuture;
import com.enterpriseagentos.backendjava.domain.common.primitives.*;
import com.enterpriseagentos.backendjava.domain.common.services.*;
import com.enterpriseagentos.backendjava.domain.common.valueobjects.*;
import com.enterpriseagentos.backendjava.domain.events.*;
import com.enterpriseagentos.backendjava.domain.features.agents.*;
import com.enterpriseagentos.backendjava.domain.features.analytics.*;
import com.enterpriseagentos.backendjava.domain.features.management.*;
import com.enterpriseagentos.backendjava.domain.features.mcp.*;

public interface IAgentToolPermissionRepository {
    CompletableFuture<List<AgentToolPermissionRecord>> listForAgentAsync(UUID agentId);
    CompletableFuture<Void> upsertAsync(UUID agentId, String skillName, String toolName, ToolPermission permission);
    CompletableFuture<Void> setManyAsync(UUID agentId, List<AgentToolPermissionRecord> entries);
}

File: apps/backend/src/Features/Agents/Domain/IAgentToolPermissionRepository.cs
namespace OffceOs.Domain.Features.Agents;

public interface IAgentToolPermissionRepository
{
    Task<IReadOnlyList<AgentToolPermissionRecord>> ListForAgentAsync(Guid agentId, CancellationToken ct = default);
    Task UpsertAsync(Guid agentId, string skillName, string toolName, ToolPermission permission, CancellationToken ct = default);
    Task SetManyAsync(Guid agentId, IReadOnlyList<AgentToolPermissionRecord> entries, CancellationToken ct = default);
}
